tests/integration/sync/test_sync_code.py

```diff
@@ -618,3 +621,59 @@ def test_sync_code_nested_getattr_layer(self):
                 lambda_response = json.loads(self._get_lambda_response(lambda_function))
                 self.assertIn("extra_message", lambda_response)
                 self.assertEqual(lambda_response.get("message"), "9")
+
+
+@skipIf(SKIP_SYNC_TESTS, "Skip sync tests in CI/CD only")
+@parameterized_class([{"dependency_layer": True}, {"dependency_layer": False}])
+class TestSyncCodeEsbuildFunctionTemplate(TestSyncCodeBase):
+    template = "template-esbuild.yaml"
+    folder = "code"
+
+    def test_sync_code_esbuild_function(self):
+        shutil.rmtree(Path(TestSyncCodeBase.temp_dir).joinpath("esbuild_function"), ignore_errors=True)
+        shutil.copytree(
+            self.test_data_path.joinpath("code").joinpath("after").joinpath("esbuild_function"),
+            Path(TestSyncCodeBase.temp_dir).joinpath("esbuild_function"),
+        )
+
+        self.stack_resources = self._get_stacks(TestSyncCodeBase.stack_name)
+        if self.dependency_layer:
+            # Test update manifest
+            layer_contents = self.get_dependency_layer_contents_from_arn(
+                self.stack_resources, str(Path("nodejs", "node_modules")), 1
+            )
+            self.assertNotIn("@faker-js", layer_contents)
+
+        # Run code sync
+        sync_command_list = self.get_sync_command_list(
+            template_file=TestSyncCodeBase.template_path,
+            code=True,
+            watch=False,
+            resource_list=["AWS::Serverless::Function"],
+            dependency_layer=self.dependency_layer,
+            stack_name=TestSyncCodeBase.stack_name,
+            parameter_overrides="Parameter=Clarity",
```

Comment:
Do you need paramater overrides, image_repository, tags, kms_key_id you could get rid of a few params here.